src/crypto/crypto.cpp

```diff
@@ -242,6 +249,126 @@ namespace crypto {
     return sc_isnonzero(&c) == 0;
   }
 
+  void crypto_ops::generate_tx_proof(const hash &prefix_hash, const public_key &R, const public_key &A, const public_key &D, const secret_key &r, signature &sig) {
+    boost::lock_guard<boost::mutex> lock(random_lock);
```

Comment:
It's there just because I wrote this function by first copying code from `generate_signature()`. I see 3 other similar uses of `boost::lock_guard` in crypto.cpp. Shall I change those as well in this PR, or make a separate PR, or leave them?